app/controllers/image-set.js

```diff
@@ -0,0 +1,73 @@
+import Ember from 'ember';
+import {imageTypes, genders} from 'lion-guardians/utils/units';
+
+export default Ember.Controller.extend({
+  imageTypes: imageTypes,
+  genders: genders,
+  uploadImageType: imageTypes[0],
+  uploadIsPublic: false,
+  isEditing: false,
+
+  organizations: function() {
+    return this.store.findAll('organization');
+  }.property(),
+
+  actions: {
+    addImage: function(upload){
+      var url = upload.get('url'),
+          isPublic = this.get('uploadIsPublic'),
+          imageType = this.get('uploadImageType'),
+          model = this.get('model');
+
+      model.addImage(url, isPublic, imageType);
+
+      Ember.run.later(this, function(){
+        // delete upload so it disappears from upload window
+        upload.deleteRecord();
+      }, 3000);
+    },
+
+    makeMainImage: function(image) {
+      this.set('model.mainImage', image);
+    },
+
+    deleteImage: function(image) {
+      var imageSet = this.get('model');
+      imageSet.get('images').removeObject(image);
+      if (imageSet.get('mainImage') === image) {
+        imageSet.set('mainImage', null);
+      }
+      image.destroy();
+    },
+
+    saveImageSet: function() {
+      var imageSet = this.get('model'),
+          controller = this;
+      imageSet.save().then(function(imageSet) {
+
+        // Can transition as soon as imageSet is saved and
+        // save the rest of the images async.
+        controller.transitionToRoute('image-set', imageSet);
+
+        imageSet.get('mainImage').save().then(function(mainImage) {
+          imageSet.save(); // save again with id for Main Image.
+
+          // Save the rest of the images
+          imageSet.get('images').forEach(function(image) {
+            if (image !== mainImage && image.get('isDirty')) {
+              image.save();
+            }
+          });
+        });
+      });
+    },
```

Comment:
This is a little risky — especially on a potentially high-latency and/or intermittent connection, making the saves of the individual images happen in the background really increases the risk that the user will close their browser tab and any images that were in the process of saving will silently be discarded.

I would save the entire image set — with all its images — in one go.
You can specify that the images should be embedded in the serialized payload for the image set, see here:
http://emberjs.com/api/data/classes/DS.EmbeddedRecordsMixin.html#stq=&stp=0

In particular you'd want to, in the image set adapter mix in the embedded records mixin and then add a section:
```
attrs: {
  images:  {serialize: 'records'}
}
```

This should make it so that when you call imageSet.save  it sends a payload like:
```
{
  image_set: {
    name: ...,
    ...,
    images: [{
      url: ...,
      ...
    }, {
      url: ...,
      ...
    }, ... ]
  }
}
```

You will need to do some trial and error to make sure that's right, but I think that's generally the gist of it.